### PR TITLE
Parse merge source as table factor

### PR DESCRIPTION
**Pull request by mskrzypkows**
Some MERGE queries need a table as a source (not just a subquery), added proper test showing it

See https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15 and https://www.postgresql.org/message-id/attachment/23520/sql-merge.html

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -1040,9 +1040,7 @@ pub enum Statement {
         // Specifies the table to merge
         table: TableFactor,
         // Specifies the table or subquery to join with the target table
-        source: Box<SetExpr>,
-        // Specifies alias to the table that is joined with target table
-        alias: Option<TableAlias>,
+        source: TableFactor,
```
Comment (mskrzypkows) [inline src/ast/mod.rs:1043]: @mobuchowski What do you think about this change?

Comment (mobuchowski) [reply src/ast/mod.rs:1043]: Looks good to me 👍

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @mskrzypkows 

This PR looks like it has a merge conflict nwo

**Comment by alamb** [comment]
FYI I plan to release 0.17.0 tomorrow in case you want to work on this PR before then: https://github.com/sqlparser-rs/sqlparser-rs/pull/488

**Comment by mskrzypkows** [comment]
> FYI I plan to release 0.17.0 tomorrow in case you want to work on this PR before then: #488
OK, conflicts are resolved

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @mskrzypkows

**Pull request merged**